lib/transform.js

```diff
@@ -4,22 +4,48 @@ var Fs = require('fs');
 
 // Declare internals
 
-var internals = {};
+var internals = {
+    fileCache: {}
+};
+
+internals.prime = function (extension, transform) {
+
+    require.extensions[extension] = function (localModule, filename) {
+
+        var src = Fs.readFileSync(filename, 'utf8');
+        var relativeFilename = filename.substr(process.cwd().length + 1);
+
+        src = (typeof transform === 'function') ? transform(src, relativeFilename) : src;
+
+        internals.fileCache[relativeFilename] = src;
 
+        localModule._compile(src, filename);
+    };
+};
 
 exports.install = function (settings) {
 
     settings.transform.forEach(function (transform) {
 
-        require.extensions[transform.ext] = function (module, filename) {
+        internals.prime(transform.ext, transform.transform);
+    });
+};
+
+exports.retrieveFile = function (path) {
```

Comment:
@geek It would be preferable yes. Now, that I look at it it seems it could be possible. I will have to move internals.transform in coverage.js and externalize it in transform.js. Maybe some other changes as well. I'll look into it.

I initially did it because I am storing all the transformed files in a cache (in internals.fileCache for transform.js and coverage.js), so that I can give it to source-map-support. Otherwise, source-map-support would read the file from the disk and it would not have the sourcemap available.